### PR TITLE
Fixes truncation warnings by replacing %s with %.s

### PR DESCRIPTION
**Pull request by nefthy**
Fixes gcc warnings about format trunctations

### CODE REVIEW

**Comment by nefthy** [comment]
@pgaskin can you review?

**Review by pgaskin — APPROVED** [review]
LGTM (build: [before](https://github.com/pgaskin/cmus/runs/1693832535?check_suite_focus=true#step:5:148), [after](https://github.com/pgaskin/cmus/runs/1697912602?check_suite_focus=true#step:5:148)), works ok.

**Comment by nefthy** [comment]
thanks. any ideas on the comment above concerning #pragma?

**Comment by pgaskin** [comment]
> thanks. any ideas on the comment above concerning #pragma?

That's due to the second parameter since GCC doesn't realize the max length 256 since `pretty` is defined in another file. To fix the warning, you can either:

- Change it to `%.256s`.
- Keep the current pragma.
- Increase the buffer size > 535.

Usually, you'd use the `alloc_size` attribute for this kind of thing, but it only works when the size is passed as a function argument (I guess one could be added and checked to ensure it's < 256, but that's a larger refactor).

File: ui_curses.c [hunk 1]
```diff
@@ -1105,8 +1105,10 @@ static void update_editable_window(struct editable *e, const char *title, const
 			utf8_encode_to_buf(filename);
 			filename = conv_buffer;
 		}
-		snprintf(buf, sizeof(buf), "%s %s - %d tracks", title,
-				pretty(filename), e->nr_tracks);
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-truncation"
+		snprintf(buf, sizeof(buf), "%s %s - %d tracks", title, pretty(filename), e->nr_tracks);
+#pragma GCC diagnostic pop
```
Comment (pgaskin) [inline ui_curses.c:1111]:
```suggestion
		snprintf(buf, sizeof(buf), "%s %.256s - %d tracks", title, pretty(filename), e->nr_tracks);
```

**Pull request merged**